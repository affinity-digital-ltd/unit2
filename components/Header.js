import { Component } from 'react'

import SubscriptionForm from '../components/SubscriptionForm'

class Header extends Component {
  render () {
    return <div className='row' style={{ marginBottom: '2rem' }}>
      <div className='col-12'>
        <div className='card'>
          <div className='card-header'>
            If you would like to receive notifications about new posts, then please consider signing up.
              </div>
          <div className='card-body'>
            <SubscriptionForm />
          </div>
        </div>
      </div>
    </div>
  }
}

export default Header
