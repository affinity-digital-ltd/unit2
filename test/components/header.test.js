import Header from '../../components/Header'

describe('<Header /> ', () => {
  it('it renders correctly', () => {
    const props = {
      signupStatus: {}
    }

    const component = shallow(
      <Header {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})

describe('<Header /> with a successful signup', () => {
  it('it renders correctly', () => {
    const props = {
      signupStatus: {
        id: 1
      }
    }

    const component = shallow(
      <Header {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})

describe('<Header /> with a failing signup', () => {
  it('it renders correctly', () => {
    const props = {
      signupStatus: {
        email: ['cant be blank']
      }
    }

    const component = shallow(
      <Header {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})

