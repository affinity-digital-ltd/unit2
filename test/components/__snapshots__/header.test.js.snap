// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> it renders correctly 1`] = `
<div
  className="row"
  style={
    Object {
      "marginBottom": "2rem",
    }
  }
>
  <div
    className="col-12"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        If you would like to receive notifications about new posts, then please consider signing up.
      </div>
      <div
        className="card-body"
      >
        <SubscriptionForm />
      </div>
    </div>
  </div>
</div>
`;
